### PR TITLE
remove k_alert API

### PR DESCRIPTION
**Pull request by pabigot**
This was a little-used feature mostly replaced by workqueues.  Remove the lingering traces in live code and the API as a whole.

### CODE REVIEW

**Review by tyrantlizard77 — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
doc changes LGTM

File: drivers/neural_net/intel_gna.c [hunk 1]
```diff
@@ -111,23 +111,27 @@ static void intel_gna_interrupt_handler(struct device *dev)
...
+	do {
+		struct intel_gna_pending_resp response;
 
-	response.callback(&response.response);
-	return 0;
+		rc = k_msgq_get(&gna->response_queue, &response, K_NO_WAIT);
+		if (rc == 0) {
+			response.callback(&response.response);
+		}
+	} while (rc == 0);
```
Comment (sathishkuttan) [inline drivers/neural_net/intel_gna.c]: Does the system workqueue call the handler once for every `k_work_submit`?
If yes, then the do-while loop is not necessary.

**Review by sathishkuttan — CHANGES_REQUESTED** [review]
Let me test `drivers/neural_net/intel_gna` with this change.

File: drivers/neural_net/intel_gna.c (hunk 1 above)
Comment (pabigot) [reply drivers/neural_net/intel_gna.c]: No, it calls it once.  The queue doesn't capture how much work is to be done, so the loop is required.

**Review by andyross — APPROVED** [review]
Definitely approve on an engineering level.  Though technically this is an API change and so should be marked deprecated for a release before actual removal.

At the same time it's a really oddball API with almost no users and the upcoming release is an LTS, so...

File: drivers/neural_net/intel_gna.c (hunk 1 above)
Comment (pabigot) [reply drivers/neural_net/intel_gna.c]: Revised.  Thanks for the review.

Comment (sathishkuttan) [reply drivers/neural_net/intel_gna.c]: Thanks for the change. I'm getting a compile error (undeclared variable.
Please rename `response.callback()` to `resp.callback()`.

Comment (pabigot) [reply drivers/neural_net/intel_gna.c]: Sorry about that; I had to shorten the variable name to conform to Zephyr's line width requirements without wrapping the condition expression and wasn't able to find a test case to build to check it.  Should be fixed now.

Comment (sathishkuttan) [reply drivers/neural_net/intel_gna.c]: I understand. It builds now.

**Review by sathishkuttan — APPROVED** [review]
Changes in `drivers/neural_net/intel_gna` LGTM.

**Comment by pabigot** [comment]
Added DNM solely because this is also flagged TSC, presumably to discuss whether the API needs to be deprecated before being removed.  From my view that's a pretty subtle indicator that it shouldn't be merged even if it's been approved and passes CI.  The DNM label is at least a more visible color.

**Comment by nashif** [comment]
TSC: no objections

**Pull request merged**